core/src/test/java/nl/nn/adapterframework/stream/MessageTest.java

```diff
@@ -95,143 +93,156 @@ protected void testAsByteArray(Message adapter) throws IOException {
+	public void testInputStreamAsReaderCaptured() throws Exception {
 		ByteArrayInputStream source = new ByteArrayInputStream(testString.getBytes("utf-8"));
 		Message adapter = new Message(source);
-		StringWriter writer = adapter.captureStream();
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		testAsReader(adapter);
 		
-		String captured = writer.toString();
+		String captured = new String(outputStream.toByteArray(), "utf-8");
 		assertEquals(testString, captured);
 	}
 	
 	@Test
-	public void testStreamAsInputSourceCaptured() throws Exception {
+	public void testInputStreamAsInputSourceCaptured() throws Exception {
 		ByteArrayInputStream source = new ByteArrayInputStream(testString.getBytes("utf-8"));
 		Message adapter = new Message(source);
-		StringWriter writer = adapter.captureStream();
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		testAsInputSource(adapter);
 		
-		String captured = writer.toString();
+		String captured = new String(outputStream.toByteArray(), "utf-8");
 		assertEquals(testString, captured);
 	}
 	
 	@Test
-	public void testStreamAsByteArrayCaptured() throws Exception {
+	public void testInputStreamAsByteArrayCaptured() throws Exception {
 		ByteArrayInputStream source = new ByteArrayInputStream(testString.getBytes("utf-8"));
 		Message adapter = new Message(source);
-		StringWriter writer = adapter.captureStream();
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		testAsByteArray(adapter);
 		
-		String captured = writer.toString();
+		String captured = new String(outputStream.toByteArray(), "utf-8");
 		assertEquals(testString, captured);
 	}
 
 	@Test
-	public void testStreamAsStringCaptured() throws Exception {
+	public void testInputStreamAsStringCaptured() throws Exception {
 		ByteArrayInputStream source = new ByteArrayInputStream(testString.getBytes("utf-8"));
 		Message adapter = new Message(source);
-		StringWriter writer = adapter.captureStream();
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		testAsString(adapter);
 		
-		String captured = writer.toString();
+		String captured = new String(outputStream.toByteArray(), "utf-8");
 		assertEquals(testString, captured);
 	}
 
 	@Test
-	public void testStreamClosedButCaptured() throws Exception {
+	public void testInputStreamClosedButCaptured() throws Exception {
 		ByteArrayInputStream source = new ByteArrayInputStream(testString.getBytes("utf-8"));
 		Message adapter = new Message(source);
-		StringWriter writer = adapter.captureStream();
-		assertNotNull(writer);
+		ByteArrayOutputStream outputStream = adapter.captureBinaryStream();
+		assertNotNull(outputStream);
 		adapter.asInputStream().close();
+		adapter.close();
```

Comment:
Dit is nu dubbelop.

Reply:
Ah, foutje, had bij andere test wel adapter.asInputStream().close() weggehaald, zal deze ook welhalen